### ISSUE TITLE
[a11y]: Modal 2.4.1: Content is not within a landmark element 

### ISSUE THREAD

**Issue by Rakesh-Addepalle** [carbon-design-system/carbon#12197]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

7.57.2

### Automated testing tool and ruleset

IBM Accessibility Tester

### Description

Create a Modal  and scan with IBM Accessibility checker:

<img width="1529" alt="Screenshot 2022-09-29 at 4 09 47 PM" src="https://user-images.githubusercontent.com/6954160/193010495-4d1f1bb7-3562-4f1e-ba07-a387cba4881a.png">
<img width="1531" alt="Screenshot 2022-09-29 at 4 09 18 PM" src="https://user-images.githubusercontent.com/6954160/193010510-9af396cd-fd93-42f5-83c1-242bc3a4e580.png">

### Reproduction/example

https://stackblitz.com/edit/github-ukaxp4?file=package.json,src%2FApp.jsx,
https://github-ukaxp4--3000.local.webcontainer.io

### Steps to reproduce

Click on the button, so that Modal will be displayed, run IBM Accessibility tester plugin to get accessibility violation results

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tw15egan** [comment]
You need to place the `Modal` inside a landmark element. Here is an updated snippet of code that resolves this violation: https://stackblitz.com/edit/github-ukaxp4-becxge?file=package.json,src%2FApp.jsx
